packages/turbo/ios/RNVisitableView.swift

```diff
@@ -10,22 +10,53 @@ import UIKit
 
 class RNVisitableView: UIView, SessionSubscriber {
   var id: UUID = UUID()
+  @objc var sessionHandle: NSString? = nil {
+    didSet {
+      validateSessionProperty(propertyName: "sessionHandle", oldValue: oldValue, newValue: sessionHandle)
+    }
+  }
+  @objc var applicationNameForUserAgent: NSString? = nil {
+    didSet {
+      validateSessionProperty(propertyName: "applicationNameForUserAgent", oldValue: oldValue, newValue: applicationNameForUserAgent)
+    }
+  }
   @objc var url: NSString = "" {
     didSet {
       if (oldValue != "") {
-        self.controller?.visitableURL = URL(string: String(url))!
-        getSession()?.visit(controller!)
+        controller.visitableURL = URL(string: String(url))!
+        turboSession.visit(controller)
       }
     }
   }
+  @objc var onMessage: RCTDirectEventBlock?
   @objc var onVisitProposal: RCTDirectEventBlock?
   @objc var onLoad: RCTDirectEventBlock?
   @objc var onVisitError: RCTDirectEventBlock?
-  @objc var sessionHandle: NSString?
-  var bridge: RCTBridge?
+  @objc var onWarning: RCTDirectEventBlock?
   
-  var controller: RNVisitableViewController?
+  var bridge: RCTBridge?
   
+  private lazy var session: RNSession = {
+    let webViewConfiguration = self.getWebViewConfiguration()
+    let session = RNSessionManager.shared.findOrCreateSession(sessionHandle: self.sessionHandle!, webViewConfiguration: webViewConfiguration)
+    if(session.configurationEquals(webViewConfiguration: webViewConfiguration)){
+      sendSessionConfigurationWarning(propertyName: "webViewConfiguration")
+    }
+    return session
+  }()
+  private lazy var turboSession: Session = session.turboSession
+  private lazy var webView: WKWebView = turboSession.webView
+    
+  lazy var controller: RNVisitableViewController = {
+    let url = URL(string: String(url))!
+    let controller = RNVisitableViewController(url: url)
```

Comment:
Wondering if we could simplify this code by _not_ passing the URL here to the init'ed controller.

Instead, the URL of the controller, would be controller by a single place (`url` setter) and we might be able to drop the conditional `if (oldValue != "") ... ` in line 25.

Reply:
1. `VisitableViewController` doesn't have a constructor with no parameters. 
2. We could just pass an URL with empty string, but when you initialize `URL` object with empty string we obtain a `nil`, and  `VisitableViewController` force-unwraps the URL parameter and it would cause the app to crash.
3. If we would like to refactor the current `lazy` initialization to a method with url parameter, we would need to change the type of the `controller` variable to Optional (Swift need all properties to be initialized with a value) and add `?` in multiple places.
4. I don't think `if (oldValue != "")` check is hacky, it is necessary when you want to change url by changing `url` prop in RN. We also use it when you get the `replace` action from Turbo. 
5. We could extract `URL(...)` to a separate property, e.g. `visitableURL`, but this variable would be an Optional (reason in 2.). It would change the implementation to this:

```swift
  private var visitableURL: URL? = nil
  @objc var url: NSString = "" {
    didSet {
      visitableURL = URL(string: url as String)
      if(visitableURL != nil){
        controller.visitableURL = visitableURL
      }
      if (oldValue != "") {
        turboSession.visit(controller)
      }
    }
  }

  lazy var controller: RNVisitableViewController = {
    let controller = RNVisitableViewController(url: visitableURL!)
    controller.delegate = self
    return controller
  }()  

  override func didMoveToWindow() {
    reactViewController().addChild(controller)
    controller.view.frame = bounds // Fixes incorrect size of the webview
    controller.didMove(toParent: reactViewController())
    addSubview(controller.view)
  }
    
```